components/brave_rewards/resources/ui/components/pageWallet.tsx

```diff
@@ -346,6 +352,258 @@ class PageWallet extends React.Component<Props, State> {
+              value: amount,
+              converted: utils.convertBalance(amount, balance.rates)
+            }
+          }
+        case 5:  // ads earnings
+          const adsDate = utils.getLongMonthName(item.date)
+          return {
+            date: adsDate || '',
+            description: getLocale('adsStatement'),
+            type: 'earningFromAdsTx' as TransactionType,
+            amount: {
+              value: amount,
+              converted: utils.convertBalance(amount, balance.rates)
+            }
+          }
+        case 8:  // tips
+          const tipsDate = `${utils.getShortMonthName(item.date)} ${utils.getDayOfMonthFromTimestampSeconds(item.date)}`
+          return {
+            date: tipsDate || '',
+            description: getLocale('oneTimeTip'),
+            type: 'donationTx' as TransactionType,
+            amount: {
+              value: amount,
+              converted: utils.convertBalance(amount, balance.rates)
+            }
+          }
+        case 16:  // monthly
+          const monthlyDate = `${utils.getShortMonthName(item.date)} ${utils.getDayOfMonthFromTimestampSeconds(item.date)}`
+          return {
+            date: monthlyDate || '',
+            description: getLocale('monthlyDonation'),
+            type: 'recurringDonationTx' as TransactionType,
+            amount: {
+              value: amount,
+              converted: utils.convertBalance(amount, balance.rates)
+            }
+          }
+        default: return {
+          date: utils.getShortMonthName(new Date().toString()),
+          description: getLocale('grantsStatement'),
+          type: 'grantsTx' as TransactionType,
+          amount: {
+            value: '0.0',
+            converted: '0.00'
+          }
+        }
+      }
+    })
+  }
+
+  selectPreviousMonthFromCurrent = () => {
+    const currentTime = new Date()
+    return `${currentTime.getFullYear()}_${currentTime.getMonth() + 1}`
+  }
+
+  getMonthlyStatementMonthsAvailable = () => {
+    const { monthlyStatement } = this.props.rewardsData
+    const monthsAvailable = monthlyStatement.monthsAvailable
+    if (!monthsAvailable) {
+      const currentTime = new Date()
+      const reportKey = `${currentTime.getFullYear()}_${currentTime.getMonth() + 1}`
+      return JSON.parse(`{ \"${reportKey}\" : \"${currentTime.toLocaleString(navigator.language, { month: 'long' })} ${currentTime.getFullYear()}\" }`)
+    }
+    let monthsJson = '{ '
+    for (let ix = 0; ix < monthsAvailable.length; ix++) {
+      const date = new Date(parseInt(monthsAvailable[ix].split('_')[0], 10), parseInt(monthsAvailable[ix].split('_')[1], 10) - 1)
+      const selectionStringValue = date.toLocaleString(navigator.language, { month: 'long' }) ;
+      (ix + 1) === monthsAvailable.length ? monthsJson += `\"${monthsAvailable[ix]}\" : \"${selectionStringValue} ${date.getFullYear()}\"` : monthsJson += `\"${monthsAvailable[ix]}\" : \"${selectionStringValue} ${date.getFullYear()}\",`
```

Comment:
Nitpick: I think this would be clearer as an if-else, just given the length of this line

Reply:
Done.